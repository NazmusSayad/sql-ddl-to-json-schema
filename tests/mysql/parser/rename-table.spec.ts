import { getParsedFormat } from '../parse-handler';
import { run } from '../runner';

run(getParsedFormat, {
  'Should rename one table.': {
    queries: ['RENAME TABLE people TO persons;', 'rename table people to persons;'],
  },

  'Should rename several tables.': {
    queries: ['rename table people to persons, homes to houses,cats to pets ,test to tests;'],
  },
});
