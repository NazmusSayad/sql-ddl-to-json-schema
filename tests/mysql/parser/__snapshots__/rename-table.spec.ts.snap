// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should rename one table. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_RENAME_TABLE",
        "def": [
          {
            "table": "people",
            "newName": "persons"
          }
        ]
      }
    }
  ]
}
`;

exports[`Should rename one table. 2`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_RENAME_TABLE",
        "def": [
          {
            "table": "people",
            "newName": "persons"
          }
        ]
      }
    }
  ]
}
`;

exports[`Should rename several tables. 1`] = `
{
  "id": "MAIN",
  "def": [
    {
      "id": "P_DDS",
      "def": {
        "id": "P_RENAME_TABLE",
        "def": [
          {
            "table": "people",
            "newName": "persons"
          },
          {
            "table": "homes",
            "newName": "houses"
          },
          {
            "table": "cats",
            "newName": "pets"
          },
          {
            "table": "test",
            "newName": "tests"
          }
        ]
      }
    }
  ]
}
`;
